invenio_app_ils/circulation/utils.py

```diff
@@ -98,7 +98,10 @@ def circulation_transaction_location_validator(transaction_location_pid):
 def circulation_transaction_user_validator(transaction_user_pid):
     """Validate that the given transaction user PID is valid."""
     if has_request_context():
-        return transaction_user_pid == str(current_user.id)
+        if current_user.is_anonymous:
+            abort()
```

Comment:
This one is correct to keep, just a small change:
```suggestion
            abort(401)
```